.github/workflows/ci.yaml

```diff
@@ -112,6 +157,7 @@ jobs:
 
   build-executables:
     name: "Build using nix"
+    needs: [populate-build-cache]
```

Comment:
This one does not need what was built in the other job. The closure of these `nix build` invocations is much smaller than what `nix develop .#ci` is providing and this job also does not benefit from the cached `~/.cabal` directory.